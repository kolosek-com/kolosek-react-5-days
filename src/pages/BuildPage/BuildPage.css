.circleci-build-page {
  max-width: 1000px;
  margin: 50px auto;
  padding: 50px;
  border: 1px solid #000000;
}
