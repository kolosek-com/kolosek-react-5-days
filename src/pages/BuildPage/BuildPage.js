import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, Link } from 'react-router-dom';

export class BuildPage extends Component {
  render() {
    const { selectedBuild } = this.props

    if (!selectedBuild) {
      this.props.history.push('/')
    }

    return (
      <div>
        <h1>CircleCI</h1>

        <h3>Build no #{selectedBuild.build_num}</h3>
        <p>Project name: {selectedBuild.reponame}</p>
        <p>Branch name: {selectedBuild.branch}</p>
        <p>Status: {selectedBuild.status}</p>
        <p>Commit message: </p>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  selectedBuild: state.builds.selectedBuild
})

export default withRouter(connect(mapStateToProps)(BuildPage))