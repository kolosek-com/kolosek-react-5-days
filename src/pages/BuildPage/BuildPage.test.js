import React from 'react';
import renderer from 'react-test-renderer';
import { BuildPage } from './BuildPage'

test('BuildPage with selectedBuild', () => {
  const selectedBuild = {
    build_num: 1,
    reponame: "Test",
    branch: "test",
    status: "Success"
  }

  const component = renderer.create(
    <BuildPage selectedBuild={selectedBuild} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
