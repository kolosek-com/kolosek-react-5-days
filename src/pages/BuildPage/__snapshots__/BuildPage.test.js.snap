// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildPage with selectedBuild 1`] = `
<div>
  <h1>
    CircleCI
  </h1>
  <h3>
    Build no #
    1
  </h3>
  <p>
    Project name: 
    Test
  </p>
  <p>
    Branch name: 
    test
  </p>
  <p>
    Status: 
    Success
  </p>
  <p>
    Commit message: 
  </p>
</div>
`;
